windows/devices/uwb/UwbConnector.cxx

```diff
@@ -633,17 +634,28 @@ UwbConnector::GetResolvedDeviceEventCallbacks()
     std::vector<std::shared_ptr<::uwb::UwbRegisteredDeviceEventCallbacks>> deviceEventCallbacks;
     deviceEventCallbacks.reserve(std::size(m_deviceEventCallbacks));
 
-    for (auto it = std::begin(m_deviceEventCallbacks); it != std::end(m_deviceEventCallbacks);) {
-        auto& deviceEventCallbackWeak = *it;
+    std::vector<uint32_t> staleCallbacks;
+
+    for (auto it = std::begin(m_deviceEventCallbacks); it != std::end(m_deviceEventCallbacks); it++) {
+        auto callbackId = *it;
+        auto result = m_deviceEventCallbacksIdMap.find(callbackId);
+        if (result == std::cend(m_deviceEventCallbacksIdMap)) {
+            PLOG_ERROR << "could not find a deviceEventCallback id in the id map when it is present in m_deviceEventCallbacks. id=" << callbackId;
+            return {};
+        }
+        auto& [id, deviceEventCallbackWeak] = *result;
         auto deviceEventCallback = deviceEventCallbackWeak.lock();
         if (deviceEventCallback != nullptr) {
             deviceEventCallbacks.push_back(std::move(deviceEventCallback));
-            it = std::next(it);
         } else {
-            it = m_deviceEventCallbacks.erase(it);
+            staleCallbacks.push_back(callbackId);

```

Comment:
That function changes the container that the for loop is iterating over, so it might not be possible to do so. (I'm unsure of exactly how the erase() function invalidates subsequent iterators)

Reply:
pushed an update that addresses the extra complexity